File: bookstore-all/bookstore_frontend/src/services/cartService.js
import config from 'config';
import {postRequest, postRequest_v2} from "../utils/ajax";
import {getBook} from "./bookService";

export function addCartProduct(customerId, bookId, bookTitle, bookImage1, bookPrice, callback) {
    const url = `http://localhost:8080/addCartProduct?customerId=${customerId}&bookId=${bookId}&bookTitle=${bookTitle}&bookImage1=${bookImage1}&bookPrice=${bookPrice}`;
    postRequest(url, {}, callback);
}

export function getCartByCustomerId(customerId, callback) {
    const url = `http://localhost:8080/getCartByCustomerId?customerId=${customerId}`;
    postRequest(url, {}, data => {
        console.log(data);
        data.map(element => {
            getBook(element.bookId, callback);
        });
    });
}

export function cleanCartByCustomerId(customerId, callback) {
    const url = `http://localhost:8080/cleanCartByCustomerId?customerId=${customerId}`;
    postRequest(url, {}, data => {
        console.log(data);
        data.map(element => {
            getBook(element.bookId, callback);
        });
    });
}

File: bookstore-all/bookstore_backend/src/main/java/com/reins/bookstore/service/CartService.java
package com.reins.bookstore.service;

import com.reins.bookstore.entity.Cart;
import org.springframework.data.repository.query.Param;

import java.math.BigDecimal;
import java.util.List;

public interface CartService {
    void addCartProduct(Integer customerId, Integer bookId, String bookTitle, String bookImage1, BigDecimal bookPrice);
    List<Cart> getCartByCustomerId(Integer customerId);
    void cleanCartByCustomerId(Integer customerId);


    ;
}
